### PR TITLE
Add mypy and tests

### PR DESCRIPTION
**Pull request by Artem4590**

### CODE REVIEW

File: tests/base.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+class BaseTestClass(TestCase):
+    """Base class for creating tests. """
+
+    _mock_api = None
+    _mock_proc = None
+
+    @staticmethod
+    def async_case(func):
+        async def wrapper():
```
Comment (Themanwhosmellslikesugar) [inline tests/base.py:17]: Need doc-string

File: tests/mock_rocket_chat.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+import argparse
```
Comment (Themanwhosmellslikesugar) [inline tests/mock_rocket_chat.py:2]: Add doc-string to module

File: tests/mock_rocket_chat.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    USER_INFO_GET_SUCCESS_RESPONSE,
+    USERS_LIST_GET_SUCCESS_RESPONSE_WITHOUT_BOT_PERMISSIONS,
+    USERS_LIST_GET_SUCCESS_RESPONSE,
+    INTERNAL_SERVER_ERROR_RESPONSE,
+)
+
+
+class MockRocketChat(BaseHTTPRequestHandler):
+    def _check_endpoint(self, endpoint):
+        if re.match(f'^{endpoint}', self.path):
```
Comment (Themanwhosmellslikesugar) [inline tests/mock_rocket_chat.py:29]: Add doc-string to `_check_endpoint` method

File: tests/mock_rocket_chat.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            return True
+        return False
+
+    def _send_headers(self):
+        self.send_header('Content-Type', 'application/json')
```
Comment (Themanwhosmellslikesugar) [inline tests/mock_rocket_chat.py:34]: Add doc-string to `_send_headers` method

File: tests/mock_rocket_chat.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+        self.end_headers()
+
+    def _send_response(self, response, status_code=200):
+        self.send_response(status_code)
```
Comment (Themanwhosmellslikesugar) [inline tests/mock_rocket_chat.py:38]: Add doc-string to `_send_response` method

File: tests/run_tests.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import unittest
+
+from tests.test_restapi import TestRestAPI
+from tests.test_meeseeks_app import TestMeeseeksApp
+from tests.test_serializers import TestContextFactory
+from tests.test_core import TestMeeseeksCore
+from tests.test_commands_base import TestCommunication, TestCommandsBase, TestDialogsBase
+from tests.test_commands_mixins import TestCommandsMixin
+from tests.test_context import TestContext, TestContextRoom, TestContextUser
```
Comment (Themanwhosmellslikesugar) [inline tests/run_tests.py:9]: Keep alphabetical order

File: tests/run_tests.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import unittest
```
Comment (Themanwhosmellslikesugar) [inline tests/run_tests.py:1]: Add doc-string to module

File: tests/server_responses.py [hunk 8]
```diff
@@ -0,0 +1,145 @@
+INTERNAL_SERVER_ERROR_RESPONSE = {
```
Comment (Themanwhosmellslikesugar) [inline tests/server_responses.py:1]: Add doc-string to module

File: meeseeks/commands/base.py [hunk 9]
```diff
@@ -2,41 +2,47 @@
...
 class Communication(MeeseeksCore, ABC):
     """Contains methods for processing incoming Rocket.Chat messages. """
 
-    def _get_response_msg(self):
+    def _get_response_msg(self) -> str:
         """Return the message sent by user without prefix. """
 
-        return self._ctx.msg.replace(f'@{settings.USER_NAME}', '', 1)
+        if isinstance(self._ctx, ChangedRoomMessageCtx):
+            return self._ctx.msg.replace(f'@{settings.USER_NAME}', '', 1)
```
Comment (Themanwhosmellslikesugar) [inline meeseeks/commands/base.py:22]: Can we annotate `self._ctx` in this class and remove these isinstance checks?

File: meeseeks/commands/base.py [hunk 10]
```diff
@@ -45,99 +51,105 @@ def _get_command_methods(self):
...
     """Contains base methods for dialogs. """
 
-    def __init__(self):
+    def __init__(self) -> None:
         super().__init__()
 
         self._dialogs_methods = self._get_dialog_methods()
 
-    def _get_dialog_methods(self):
+    def _get_dialog_methods(self) -> list:
```
Comment (Themanwhosmellslikesugar) [inline meeseeks/commands/base.py:141]: List of what?

File: meeseeks/type.py [hunk 11]
```diff
@@ -0,0 +1,50 @@
+"""Module contains description of types for Meeseeks. """
```
Comment (Themanwhosmellslikesugar) [inline meeseeks/type.py:1]: Rename module to `types.py`

File: meeseeks/commands/base.py [hunk 12]
```diff
@@ -45,99 +51,105 @@ def _get_command_methods(self):
...
-        allow = False
-        if not roles:
-            allow = True
-        elif any(role in roles for role in users_info['roles']):
-            allow = True
+        allow: bool = False
+        if isinstance(self._ctx, ChangedRoomMessageCtx):
+            users_info: UserInfo | None = await self._restapi.get_user_info(self._ctx.user.id)
+            if not roles:
+                allow = True
```
Comment (Themanwhosmellslikesugar) [inline meeseeks/commands/base.py:76]: Why we allow if there are no roles?

Comment (Artem4590) [reply meeseeks/commands/base.py:76]: If there are no roles, we allow execution for all users. If we check the role it means the user must have current roles

File: .github/workflows/linter.yml [hunk 13]
```diff
@@ -25,9 +25,25 @@ jobs:
...
+
       - name: Install dependencies
         run: pip install -r requirements.txt
 
       - name: Run pylint
-        run: |
-             find -name "*.py" -not -path "./.git/*" -not -path "*/alembic/versions/*" | xargs pylint --rcfile=.pylintrc
+        run: find -name "*.py" -not -path "./.git/*" -not -path "*/alembic/versions/*" -not -path "*/tests/*" | xargs pylint --rcfile=.pylintrc
+      - name: Run mypy
+        run: find meeseeks/ -name "*.py" -not -path "./.git/*" -not -path "./tests/*" | xargs mypy
```
Comment (Themanwhosmellslikesugar) [inline .github/workflows/linter.yml:37]: Лучше запускай mypy без find
```suggestion
        run: mypy --exclude=tests/ . 
```

File: tests/test_commands_base.py [hunk 14]
```diff
@@ -0,0 +1,108 @@
+from unittest import mock
+
+from meeseeks import settings, RestAPI, MeeseeksBaseApp
+from meeseeks.commands.base import CommandsBase, Communication
+from meeseeks.context import ChangedRoomMessageCtx
+from tests.base import BaseTestClass
+
+
+class TestCommunication(BaseTestClass):
+    def test_get_response_msg(self):
```
Comment (Themanwhosmellslikesugar) [inline tests/test_commands_base.py:10]: Здесь и дальше везде по тестам нужны doc-string'и

**Pull request merged**